frontend/packages/integration-tests-cypress/tests/crud/add-storage-crud.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { testName, checkErrors } from '../../support';
+import { listPage } from '../../views/list-page';
+import { detailsPage } from '../../views/details-page';
+import { errorMessage, submitButton } from '../../views/form';
+
+const k8sWorkloads = [
+  'replicationcontrollers',
+  'daemonsets',
+  'deployments',
+  'replicasets',
+  'statefulsets',
+];
+const openshiftWorkloads = ['deploymentconfigs'];
+const resourceObjs =
+  Cypress.env('openshift') === true ? k8sWorkloads.concat(openshiftWorkloads) : k8sWorkloads;
+
+describe('Add storage is applicable for all workloads', () => {
+  before(() => {
+    cy.login();
+    cy.createProject(testName);
+  });
+
+  afterEach(() => {
+    checkErrors();
+  });
+
+  after(() => {
+    resourceObjs.forEach((resourceType) => {
+      try {
+        cy.exec(`kubectl delete --cascade ${resourceType} example -n ${testName}`, {
+          failOnNonZeroExit: false,
+        });
+      } catch (error) {
+        console.error(`Failed to delete ${resourceType} example:\n${error}`);
+      }
+    });
+    cy.deleteProject(testName);
+    cy.logout();
+  });
+
+  resourceObjs.forEach((resourceType) => {
+    const pvcName = `${resourceType}-pvc`;
+    const pvcSize = '1';
+    const mountPath = '/data';
+    describe(resourceType, () => {
+      it(`create a ${resourceType} resource and adds storage to it`, () => {
+        listPage.createNamespacedResourceWithDefaultYAML(resourceType, testName);
+        cy.get(errorMessage).should('not.exist');
+
+        detailsPage.clickPageActionFromDropdown('Add storage');
+        cy.contains('Select claim').should('be.visible');
+        cy.byTestID('Create new claim-radio-input').click();
+        cy.byTestID('pvc-name').type(pvcName);
+        cy.byTestID('pvc-size').type(pvcSize);
+        cy.byTestID('mount-path').type(mountPath);
+        cy.get(submitButton).click();
+        cy.get(errorMessage).should('not.exist');
+        cy.contains(pvcName).should('exist');
+        cy.contains(mountPath).should('exist');
```

Comment:
This just checks if the text is anywhere on the page? I don't think the check is specific enough since we could still be on the form page in theory, which would have both of these strings. Can we scope the check to test specifically that it's part of the volumes table on the workload details page?

Reply:
Good point, added a `.parent` and `.within` to limit scope:
![image](https://user-images.githubusercontent.com/12733153/118164101-b4191e00-b3f0-11eb-9057-37210c5ec0ec.png)

Reply:
Is there a reason we can't use the existing data-test attributes on the volume table columns?

Reply:
Wow!  Didn't realize we pushed `data-test-...` id's to each table cell, that does make things easier :-)
Updated to:
```
cy.get(`[data-test-volume-name-for="${pvcName}"]`).should('have.text', pvcName);
cy.get(`[data-test-mount-path-for="${pvcName}"]`).should('have.text', mountPath);
```